Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/storage/RevocationRepository.kt

```diff
@@ -0,0 +1,75 @@
+package de.rki.coronawarnapp.covidcertificate.revocation.storage
+
+import androidx.annotation.VisibleForTesting
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import de.rki.coronawarnapp.covidcertificate.revocation.RevocationDataStore
+import de.rki.coronawarnapp.covidcertificate.revocation.RevocationReset
+import de.rki.coronawarnapp.covidcertificate.revocation.model.CachedRevocationChunk
+import de.rki.coronawarnapp.tag
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.flow.shareLatest
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RevocationRepository @Inject constructor(
+    @AppScope appScope: CoroutineScope,
+    @BaseJackson private val objectMapper: ObjectMapper,
+    @RevocationDataStore private val dataStore: DataStore<Preferences>
+) {
+
+    val revocationList: Flow<List<CachedRevocationChunk>> = dataStore.data
+        .catch {
+            Timber.tag(TAG).e(it, "Failed to read DataStore")
+            if (it is IOException) emit(emptyPreferences()) else throw it
+        }
+        .map { prefs ->
+            prefs[CACHED_REVOCATION_CHUNKS_KEY]?.toCachedRevocationChunks() ?: emptyList()
+        }
+        .shareLatest(
+            tag = TAG,
```

Comment:
We dont really need to log this. But keeping shareLatest is beneficial because it makes the flow distinctUntilChanged and conflated and in the future we may get new subscribers.